src/Our.Umbraco.Vorto/Web/UI/App_Plugins/Vorto/js/vorto.js

```diff
@@ -192,6 +202,10 @@
             }
         });
 
+        $scope.$watch("model.value.values", function () {
```

Comment:
So, you mean I should put the `detectFilledInLanguages()` inside the `setActiveLanguage` function?